### PR TITLE
Fix .js file activation issue

### PR DESCRIPTION
**Pull request by frogcjn**
(1) update dependence
(2) change language activation to `javascript`, use `javascript` instead of the fake `flow` language
(3) small fixs in `launch.json`, `.gitignore`

### CODE REVIEW

**Comment by nmote** [comment]
Hmm, unfortunately this won't solve #14 since we stopped publishing the `nuclide-flow-base` package separately from Nuclide. But updating certainly won't hurt.

@bpasero What do you think about changing the language to `javascript`? It would solve some of the issues reported here but I'm not sure if it would cause interference with e.g. TypeScript. You had some motivation to create a separate Flow language, right? Have things changed since then?

**Comment by bpasero** [comment]
@nmote sorry, I am currently not able to look into flow for VS Code and have not tracked the issues here.  If we want to continue the flow extension I would hope the community to take this project over.

**Comment by nmote** [comment]
@frogcjn Can you explain the motivation for moving away from the `flow` language? Does VSCode complain when you use type annotations when the language is set to `javascript`? IIRC it used to.

**Comment by frogcjn** [comment]
@nmote 
I've checked other plugin, like `ESLint`, they just use the javascript as the language ID.
I've tries many test case, only this way can solve the activation problem.

If use a new language ID, called `flow`,
   (1) If you open a *.txt file or other extension, VSCode will automatically recognize it to Flow. It works fine.
   (2) when you open a `*.js` file, VSCode will not automatically recognize it to `flow`, but `JS`, so any capability with `flow` will be disabled, until the user tap right-bottom corner to choose.

If use the `javascript` language ID, and change its alias to `flow`, then all *.js files no matter whether its head begin with `@flow`, will be recognized to `flow`.

So if we don't modify VSCode, the only way to solve the problem is to use `javascript` language ID, and do not change its alias. Let flow-bin to decide whether it is a flow Javascript file and verify it or not.

`ESLint` use this solution.

Thanks.

**Comment by nmote** [comment]
The difference with `ESLint` is that it works with stock JavaScript syntax. For

**Comment by frogcjn** [comment]
I use the setting with VSCode:
"javascript.validate.syntaxValidation": false

**Comment by nmote** [comment]
Great, could you just update the README to note that this setting is recommended, and then I'll merge?

**Comment by frogcjn** [comment]
So there is a two choice:
(1) keep `flow` language id, and let user to choose it when open a `.js` file, because VSCode will not recognize `.js` file with `flow`.
(2) use `javascript` language id, then `"javascript.validate.syntaxValidation": false` is needed.

**Comment by nmote** [comment]
I prefer (2), but we need to make people aware of this setting.

**Comment by nmote** [comment]
Also please change the title -- I do not believe this will make the plugin work for Flow v0.25.

**Comment by frogcjn** [comment]
@nmote 
I have tried 0.0.2 existed on marketplace, if `"javascript.validate.syntaxValidation": true` and even if I choose `Flow` as the language, the `'property declarations' can only be used in a .ts file.` error still existed.

I just want to make sure the choice is right.

**Comment by frogcjn** [comment]
My flow version is 0.25, and its binary file is download from the release of Facebook/flow.

**Comment by nmote** [comment]
And do errors correctly display in the editor?

**Comment by frogcjn** [comment]
Yes.

screenshot 2016-05-25

**Comment by frogcjn** [comment]
I have figure out how to use `flow` as default language id for `.js` file, just write
` "files.associations": {"*.js": "flow"}, `

But It is still better to use language id `javascript`, because if a `.js` file choose a new language id `flow`, then `ESLint` will never display any error on that file.

File: README.md [hunk 1]
```diff
@@ -21,7 +22,7 @@ Follow the [instructions](https://code.visualstudio.com/docs/editor/extension-ga
 
 ## Known Issues
 
-* when you create a new file and add the `/* @flow */` to the beginning you need to manually change the mode from the status bar to `Flow`
+* You should set workspace preference to disabe default syntax validation from Visual Studio Code.
```
Comment (nmote) [inline README.md]: Typo in "disable".

Also, can you be a little bit more specific about how to do this? In the PR comments you said you need to set:

`"javascript.validate.syntaxValidation": true`

I think it's worth putting that down explicitly.

**Comment by nmote** [comment]
It sounds like VSCode will make this inconvenient no matter what we do. I think the best thing to do is to recommend that people disable syntax validation because of the shortcomings you've identified with having a separate Flow language.

File: README.md (hunk 1 above)
Comment (frogcjn) [inline README.md]: OK.

**Comment by frogcjn** [comment]
Fixed.

**Comment by nmote** [comment]
Thanks! I'll publish this week.

**Comment by nmote** [comment]
Hey, things got busy but I'll do this on Tuesday. Sorry!

**Comment by nmote** [comment]
Just published 0.0.3: https://marketplace.visualstudio.com/items?itemName=flowtype.flow-for-vscode

**Comment by frogcjn** [comment]
OK. Thanks!

**Pull request merged**